src/org/zaproxy/zap/extension/api/API.java

```
@@ -440,6 +455,16 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 			logger.debug("handleApiRequest returning: " + response);
 			
 		} catch (Exception e) {
+			if (! getOptionsParamApi().isReportPermErrors()) {
+				if (e instanceof ApiException) {
+					ApiException exception = (ApiException) e;
+					if (exception.getType().equals(ApiException.Type.BAD_API_KEY) ||
+							exception.getType().equals(ApiException.Type.BAD_API_KEY)) {
+						// Fail silently
```

Should also fail silently if disabled?